axlearn/audio/asr_decoder.py

```diff
@@ -28,6 +28,40 @@
 from axlearn.common.utils import Nested, Tensor
 
 
+def _is_valid_ctc_seq(paddings: Tensor, target_labels: Tensor, target_paddings: Tensor) -> Tensor:
+    """Returns for per example sequence if it passes validity check.
+
+    Note that `optax.ctc_loss` returns -logeps (default to 1e5) if the
+    input length is smaller than the label length plus number of
+    consectutive duplications, because we need an blank to token to transit
+    between the same tokens. When this condition is not met, it should be
+    considered as an invalid sequence and the loss should be ignored.
```

Comment:
```suggestion
    consecutive duplications, because we need a blank to token to transition
    between the same labels. When this condition is not met, it should be
    considered as an invalid sequence and the loss should be ignored.
```